### DISCUSSION TITLE
Issue with Vite (Outdated optimize dep)

### DISCUSSION THREAD

**Discussion by KirillRotanov** [KevinVandy/material-react-table#982] [Q&A]
Hi, fellas!
I encountered a problem in my React TS Vite project,
the development of the MRT as a separate component was carried out in a storybook and there were no problems. But as soon as I decided to integrate the table into the application, I started getting strange errors:

<img width="608" alt="Screenshot 2024-02-08 at 15 34 43" src="https://github.com/KevinVandy/material-react-table/assets/70111639/ef5b2c78-c1e2-4c0c-9675-4702af5ba319">

or

<img width="890" alt="Screenshot 2024-02-08 at 16 33 12" src="https://github.com/KevinVandy/material-react-table/assets/70111639/334f4fb9-922e-442c-a364-3e19a9f69af9">

<img width="1114" alt="Screenshot 2024-02-08 at 16 35 10" src="https://github.com/KevinVandy/material-react-table/assets/70111639/d1e28704-932d-4442-b8ef-2cf5aa99db92">

I went here (https://github.com/mui/material-ui/issues/31835) to find the solution, but all that i found it's to change the imports from " import Box from '@mui/material/Box' " to " import { Box } from '@mui/material' ".

Since i can't do it inside the MRT node module, I went to find another solution and found these branches on Vite's github (https://github.com/vitejs/vite/issues/12434  https://github.com/vitejs/vite/issues/8308) and ofcause a ton of questions on stackoverflow.

so I tried to change optimizeDeps option in vite.config.ts and in case of

optimizeDeps: {
    include: ['@emotion/styled'],
    exclude: [
       'material-react-table',
    ],
}

I'm getting this error 
<img width="612" alt="Screenshot 2024-02-08 at 16 16 21" src="https://github.com/KevinVandy/material-react-table/assets/70111639/26fb3793-73d6-407d-b9d8-a9fed58cbbf9">

in case of 

optimizeDeps: {
    include: ['@emotion/styled'],
    exclude: [
       '@mui/material',
    ],
}

getting this
<img width="610" alt="Screenshot 2024-02-08 at 16 18 36" src="https://github.com/KevinVandy/material-react-table/assets/70111639/40656b59-c9c7-40ad-9e73-d480c0601287">

Unfortunately none of the suggested solutions worked for me.. 

this change in vite.config.ts doesn't help:

build: {
     target: 'esnext',
...other options,
}

also as 

optimizeDeps: {
    disabled: true
}

NOTE: I deleted and did reinstall to all the node_modules on each interaction with vite.config.ts and ran project with the --force flag
stack is:
    "react": "18.2.0",
    "vite": "5.0.12",
    "material-react-table": "2.11.1",
    "@mui/material": "5.15.9",

So I actually don't know what else I can try to fix this issue..
Will be very grateful for any help

**Comment by KevinVandy** [comment]
Would you be able to fork the basic example in code sandbox, and provide your package.json and vite config to reproduce this error?

**Reply by KirillRotanov** [reply]
https://stackblitz.com/edit/github-ndie5t?file=package.json,vite.config.js
the thing is that it works... I'll try to implement my logic, to see if it affects somehow on this issue. I'll let you know

**Comment by githorse** [comment]
I don't know anything about this specific instance, but the `--force` flag has been my goto workaround for the "Outdated optimize dep" issue, which I get a lot:

```
$ vite serve --port 3000 --force
```
